src/debugpy/_vendored/pydevd/pydevd.py

```diff
@@ -1317,7 +1317,11 @@ def has_user_threads_alive(self):
                     'Error in debugger: Found PyDBDaemonThread not marked with is_pydev_daemon_thread=True.\n')
 
             if is_thread_alive(t):
-                if not t.isDaemon() or hasattr(t, "__pydevd_main_thread"):
+                try:
+                    is_daemon = t.daemon
```

Comment:
`t.daemon` is available in `Python 2.6` already, so, I don't think we need the try..except.